vars/withSubscription.groovy

```diff
@@ -90,7 +90,7 @@ def identityBasedLogin(String subscription, Closure body) {
     withAzureKeyvault([
       [$class: 'AzureKeyVaultSecret', secretType: 'Secret', name: "${subscription}-subscription-id", version: '', envVariable: 'ARM_SUBSCRIPTION_ID']
     ]) {
-      az "account set --subscription ${env.ARM_SUBSCRIPTION_ID}"
+      az "account set --subscription ${env.JENKINS_SUBSCRIPTION_NAME}"
```

Comment:
no you should do this on the azJenkins, `az` is for the subscription you're currently working in
```suggestion
      az "account set --subscription ${env.ARM_SUBSCRIPTION_ID}"
      azJenkins "account set --subscription ${env.JENKINS_SUBSCRIPTION_NAME}"
```

They use different AZURE_CONFIG_DIR so that it can have different accounts set